dags/glam_dev.py

```diff
@@ -0,0 +1,445 @@
+
+from glam_subdags.extract import extracts_subdag, extract_user_counts
+from glam_subdags.histograms import histogram_aggregates_subdag
+from glam_subdags.general import repeated_subdag
+from glam_subdags.generate_query import generate_and_run_desktop_query
+from utils.gcp import bigquery_etl_query, gke_command
+from utils.tags import Tag
+
+
+prod_project_id = "moz-fx-data-shared-prod"
+prod_dataset_id = "telemetry_derived"
+dev_dataset_id = "dev_telemetry_derived"
+tmp_project = "moz-fx-data-shared-prod"  # for temporary tables in analysis dataset
+default_args = {
+    "owner": "efilho@mozilla.com",
+    "depends_on_past": False,
+    "start_date": datetime(2019, 10, 22),
+    "email": [
+        "efilho@mozilla.com",
+    ],
+    "email_on_failure": True,
+    "email_on_retry": True,
+    "retries": 1,
+    "retry_delay": timedelta(minutes=30),
+}
+
+GLAM_DAG = "glam-dev"
+GLAM_CLIENTS_HISTOGRAM_AGGREGATES_SUBDAG = "clients_histogram_aggregates"
+PERCENT_RELEASE_WINDOWS_SAMPLING = "10"
+
+tags = [Tag.ImpactTier.tier_2]
+
+dag = DAG(
+    GLAM_DAG,
+    default_args=default_args,
+    schedule_interval=None,
+    doc_md=__doc__,
+    tags=tags,
+)
+
+""" This isn't needed because the dev dag will only be triggered manually
+# Make sure all the data for the given day has arrived before running.
+wait_for_main_ping = ExternalTaskCompletedSensor(
+    task_id="wait_for_main_ping",
+    external_dag_id="copy_deduplicate",
+    external_task_id="copy_deduplicate_main_ping",
+    execution_delta=timedelta(hours=1),
+    check_existence=True,
+    mode="reschedule",
+    pool="DATA_ENG_EXTERNALTASKSENSOR",
+    email_on_retry=False,
+    dag=dag,
+    docker_image="gcr.io/moz-fx-data-airflow-prod-88e0/glam-dev-bigquery-etl:latest",
+)
+"""
+latest_versions = bigquery_etl_query(
+    task_id="latest_versions",
+    destination_table="latest_versions",
+    dataset_id=dev_dataset_id,
+    project_id=prod_project_id,
+    owner="efilho@mozilla.com",
+    date_partition_parameter=None,
+    arguments=("--replace",),
+    dag=dag,
+)
+
+# This task runs first and replaces the relevant partition, followed
+# by the next two tasks that append to the same partition of the same table.
+clients_daily_scalar_aggregates = generate_and_run_desktop_query(
+    task_id="clients_daily_scalar_aggregates",
+    project_id=prod_project_id,
+    source_dataset_id=prod_dataset_id,
+    destination_dataset_id=dev_dataset_id,
+    sample_size=PERCENT_RELEASE_WINDOWS_SAMPLING,
+    overwrite=True,
+    probe_type="scalar",
+    dag=dag,
+    docker_image="gcr.io/moz-fx-data-airflow-prod-88e0/glam-dev-bigquery-etl:latest",
+)
+
+clients_daily_keyed_scalar_aggregates = generate_and_run_desktop_query(
+    task_id="clients_daily_keyed_scalar_aggregates",
+    project_id=prod_project_id,
+    source_dataset_id=prod_dataset_id,
+    destination_dataset_id=dev_dataset_id,
+    sample_size=PERCENT_RELEASE_WINDOWS_SAMPLING,
+    overwrite=False,
```

Comment:
From what I understand this table will keep appending data. If so, should we allow dev tables to keep appending and grow large over time?

Reply:
I'd say yes. We'll probably delete those tables after testing is finished in dev, but during testing we want to have things happen as similar to Prod as we can.

Reply:
+1, don't worry about the table size.
+1 for making sure to clean this up afterwards.

Reply:
cleanup is being tracked here https://github.com/mozilla/glam/issues/2177